MonoGame.Framework/Content/ContentReaders/EffectReader.cs

```diff
@@ -31,11 +36,19 @@ private static bool Contains(string search, string[] arr)
 
         protected internal override Effect Read(ContentReader input, Effect existingInstance)
         {
-            int count = input.ReadInt32();
+            Effect effect = existingInstance;
+
+            var dataSize = (int)input.ReadUInt32();
+            byte[] data = MemoryPool.Current.GetPooledBuffer(dataSize);
+            input.Read(data, 0, dataSize);
+
+            if (effect == null)
+            {
+                effect = new Effect(input.GraphicsDevice, data, 0, dataSize);
+                MemoryPool.Current.PoolBuffer(data);
```

Comment:
This looks like a memory leak.  If `effect != null` the `data` is never returned the to pool.